geospaas_harvesting/crawlers.py

```diff
@@ -231,12 +250,44 @@ def _get_links(cls, html):
         return parser.links
 
 
+class OpenDAPCrawler(WebDirectoryCrawler):
+    """
+    Crawler for harvesting the data of OpenDAP
+    """
+    LOGGER = logging.getLogger(__name__ + '.OpenDAPCrawler')
+    FOLDERS_SUFFIXES = ('/contents.html',)
+    FILES_SUFFIXES = ('.nc', '.nc.gz')
+    EXCLUDE = ['?']
+
+    def get_download_url(self, resource_url):
+        return resource_url
+
+
+class ThreddsCrawler(WebDirectoryCrawler):
+    """
+    Crawler for harvesting the data which are provided by OpenDAP
+    """
+    LOGGER = logging.getLogger(__name__ + '.ThreddsCrawler')
+    FOLDERS_SUFFIXES = ('/catalog.html',)
+    FILES_SUFFIXES = ('.nc',)
+    EXCLUDE = ['/thredds/', 'http',]
+    def get_download_url(self, resource_url):
+        links = self._get_links(self._http_get(resource_url))
+        for link in links:
+            if "dodsC" in link:
+                if not link.endswith(".html"):
+                    self.LOGGER.warning('The link as the result of crawler for "%s" must be ended with ".html".Failed to create downloable form',resource_url)
+                result = "https://thredds.met.no"+link[:-4]+'dods'
+                return result
+            else:
+                return None
```

Comment:
it would be a bit more elegant to initialize `result = None` before the loop, then just return `result`

Reply:
addressed in 3dec7fe